src/main/webapp/qaReport/QaReport.js

```diff
@@ -0,0 +1,269 @@
+      projectType: this.state.projectType.value
+    };
+
+    spinnerService.show(QA_REPORT_SPINNER);
+    try {
+      let result = await Reports.getQaEventReport(query, tab, filter);
+      const lastPage = Math.ceil(result.data.recordsFiltered / query.length);
+      this.setState(prev => {
+        prev.query = query;
+        prev[tab].lastPage = lastPage;
+        prev[tab].currentPage = page;
+        prev[tab].data = result.data.data;
+        prev[tab].recordsTotal = result.data.recordsTotal;
+        prev[tab].recordsFiltered = result.data.recordsFiltered;
+        prev[tab].sizePerPage = query.length;
+        prev[tab].search = query.searchValue;
+        prev[tab].sort = {
+          orderColumn : query.orderColumn,
+          sortDirection: query.sortDirection
+        };
+        prev[tab].hide = result.data.data.length === 0;
+        prev.activeTab = prev[IRB].hide ? NO_IRB : IRB;
+        return prev;
+      }, () => {
+        spinnerService.hide(QA_REPORT_SPINNER)
+      })
+    } catch(error) {
+      spinnerService.hide(QA_REPORT_SPINNER);
+      this.setState(() => { throw error });
+    }
+  };
+
+  onTableChange = async (type, newState, tab) => {
+    switch(type) {
+      case TABLE_ACTIONS.SEARCH: {
+        await this.onSearchChange(newState.searchText, tab);
+        break
+      }
+      case TABLE_ACTIONS.FILTER: {
+        // Not implemented
+        break;
+      }
+      case TABLE_ACTIONS.PAGINATION: {
+        await this.onPageChange(newState.page, newState.sizePerPage, tab);
+        break;
+      }
+      case TABLE_ACTIONS.SORT: {
+        await this.onSortChange(newState.sortField, newState.sortOrder, tab);
+        break;
+      }
+    }
+  };
+
+  onSearchChange = async (search, tab) => {
+    await this.tableHandler(0, this.state[tab].sizePerPage, search, this.state[tab].sort, 1, tab);
+  };
+
+  onPageChange = async (page, sizePerPage, tab) => {
+    const offset = (page - 1) * sizePerPage;
+    await this.tableHandler(offset, sizePerPage, this.state[tab].search, this.state[tab].sort, page, tab);
+  };
+
+  onSortChange = async (sortName, sortOrder, tab) => {
+    const sort = {
+      sortDirection: sortOrder,
+      orderColumn: QA_EVENT_SORT_NAME_INDEX[sortName]
+    };
+    this.tableHandler(0, this.state[tab].sizePerPage, null, sort,0 , tab)
+  };
+
+  handleDatePicker = (e) => (date) => {
+    this.setState(prev => {
+      prev[e] = date;
+      return prev;
+    }) ;
+  };
+
+  handleSelectProjectType = (e) => {
+    this.setState(prev => {
+      prev.projectType = e;
+      return prev;
+    }) ;
+  };
+
+  applyFilterPanel = async () => {
+    if (this.state.projectType.value === 'irb') {
+      this.setState(prev => {
+        prev.activeTab = IRB;
+        prev[NO_IRB].hide = true;
+        }, async () =>
+        await this.tableHandler(0, this.state[IRB].sizePerPage, this.state[IRB].search, this.state[IRB].sort, this.state[IRB].currentPage, IRB)
```

Comment:
I think it would be more readable if this can be split in multiple lines.
```suggestion
        await this.tableHandler(
          0, 
          this.state[IRB].sizePerPage, 
          this.state[IRB].search, 
          this.state[IRB].sort, 
          this.state[IRB].currentPage, 
          IRB
        )
      );
```